Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,390 @@
+function [PWI, PWI3D, PWI4D, x, Control, Control3D, Control4D] = xASL_im_ASLSubtractionAveraging(x, path_ASL4D, PWI4D, PWI3D)
+%xASL_im_ASLSubtractionAveraging Central function for ASL subtraction and averaging, for both PWI & controls
```

Comment:
I've detected a major bug!

You have written the function in a way that you can provide PWI4D on input instead of calculating it inside. Which is fine and I fully support that.

But in case PWI4D is provided, then PWI4D is not calculated, and then PWI3D is calculated from PWI4D. But at the same time, Control3D is calculated from Control4D, which is missing on the input.

So you have to add Control4D and Control3D also on the input, you have to change all calls of this function everywhere, update header, update all the admin and calculations. And the entire function call might become a bit hard to understand - we might need to redefine the whole concept.

Note that I'm not against this concept, just reporting this serious bug and noting that incorporating Control images might make it a bit complex on the input side.

Reply:
I see, so the solution is that Control3D is only calculated from Control4D, and PWI3D only from PWI4D.
So I added to the issue:

- [ ] ASL4D -> PWI4D -> PWI3D -> PWI is separated from ASL4D -> Control4D -> Control3D -> Control

Reply:
Yes - that is one possible solution. You just have to make sure that Control4D is optionally on the input. And that it has ASL4D when not. And that the various combinations are properly checked on the input and that there can be some empty inputs that are skipped. The input checking wasn't really good and woudlnt' work with that. I added some fixes, but more are needed.

Reply:
Done